src/Evolution/Systems/Cce/AnalyticSolutions/RobinsonTrautman.cpp

```diff
@@ -0,0 +1,471 @@
+      square(extraction_radius_) * real(get(local_bondi_u).data());
+  get<1, 2>(*spherical_metric) =
+      -square(extraction_radius_) * real(get(local_bondi_u).data());
+  // note: using 'pfaffian' components, because otherwise the sin(theta)s lose
+  // precision at the poles
+  get<0, 3>(*spherical_metric) =
+      square(extraction_radius_) * imag(get(local_bondi_u).data());
+  get<1, 3>(*spherical_metric) =
+      -square(extraction_radius_) * imag(get(local_bondi_u).data());
+
+  // note: pfaffian components
+  get<2, 2>(*spherical_metric) = square(extraction_radius_);
+  get<2, 3>(*spherical_metric) = 0.0;
+  get<3, 3>(*spherical_metric) = square(extraction_radius_);
+}
+
+void RobinsonTrautman::dr_spherical_metric(
+    const gsl::not_null<
+        tnsr::aa<DataVector, 3, ::Frame::Spherical<::Frame::Inertial>>*>
+        dr_spherical_metric,
+    const size_t l_max, const double time) const noexcept {
+  ASSERT(time == prepared_time_,
+         "The Robinson-Trautman solution is being calculated in an "
+         "inconsistent state. The public interface should always call "
+         "`prepare_solution` before calculating any metric components, so this "
+         "may indicate inconsistent use of the internal Robinson-Trautman "
+         "calculation functions.");
+  ASSERT(l_max == l_max_,
+         "The Robinson-Trautman solution only supports the l_max resolution "
+         "specified at construction, as it must internally store the evolved "
+         "scalar at the desired resolution");
+  Variables<tmpl::list<Tags::BondiW, Tags::Dr<Tags::BondiW>, Tags::BondiU>>
+      temporary_variables{
+          Spectral::Swsh::number_of_swsh_collocation_points(l_max_)};
+  auto& local_dr_bondi_w = get<Tags::Dr<Tags::BondiW>>(temporary_variables);
+  auto& local_bondi_w = get<Tags::BondiW>(temporary_variables);
+  auto& local_bondi_u = get<Tags::BondiU>(temporary_variables);
+
+  bondi_u(make_not_null(&local_bondi_u), dense_output_rt_scalar_);
+  dr_bondi_w(make_not_null(&local_dr_bondi_w), dense_output_rt_scalar_);
+  bondi_w(make_not_null(&local_bondi_w), dense_output_rt_scalar_);
+
+  dt_spherical_metric(dr_spherical_metric, l_max, time);
+
+  get<0, 0>(*dr_spherical_metric) =
+      -get<0, 0>(*dr_spherical_metric) -
+      real(get(dense_output_rt_scalar_).data() *
+           (extraction_radius_ * get(local_dr_bondi_w).data() +
+            get(local_bondi_w).data()));
+  get<0, 1>(*dr_spherical_metric) =
+      real(get(dense_output_rt_scalar_).data() *
+           (extraction_radius_ * get(local_dr_bondi_w).data() +
+            get(local_bondi_w).data())) -
+      get<0, 1>(*dr_spherical_metric);
+  get<1, 1>(*dr_spherical_metric) =
+      -real(get(dense_output_rt_scalar_).data() *
+            (extraction_radius_ * get(local_dr_bondi_w).data() +
+             get(local_bondi_w).data())) -
+      get<1, 1>(*dr_spherical_metric);
+  // note: bondi_u is proportional to 1/r, so when comparing to the values in
```

Comment:
Can we move this comment to the top of these computations? Makes the earlier components more obvious.